homeassistant/components/tradfri/sensor.py

```diff
@@ -27,43 +39,82 @@ async def async_setup_entry(
     coordinator_data = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
     api = coordinator_data[KEY_API]
 
-    async_add_entities(
-        TradfriBatterySensor(
-            device_coordinator,
-            api,
-            gateway_id,
-        )
-        for device_coordinator in coordinator_data[COORDINATOR_LIST]
+    entity_list: list[TradfriSensor] = []
+
+    for device_coordinator in coordinator_data[COORDINATOR_LIST]:
         if (
             not device_coordinator.device.has_light_control
             and not device_coordinator.device.has_socket_control
             and not device_coordinator.device.has_signal_repeater_control
             and not device_coordinator.device.has_air_purifier_control
-        )
-    )
+        ):
+            entity_list.append(
+                TradfriSensor(
+                    device_coordinator,
+                    api,
+                    gateway_id,
+                    description=TradfriSensorEntityDescription(
+                        device_class=SensorDeviceClass.BATTERY,
+                        native_unit_of_measurement=PERCENTAGE,
+                        key=SensorDeviceClass.BATTERY,
+                        value=lambda data: cast(
+                            int, data.coordinator.data.device_info.battery_level
+                        ),
+                    ),
+                )
+            )
+        elif device_coordinator.device.has_air_purifier_control:
+            entity_list.append(
+                TradfriSensor(
+                    device_coordinator,
+                    api,
+                    gateway_id,
+                    description=TradfriSensorEntityDescription(
+                        device_class=SensorDeviceClass.AQI,
+                        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+                        key=SensorDeviceClass.AQI,
+                        # The sensor returns 65535 if the fan is turned off
+                        value=lambda data: None
+                        if data.coordinator.data.air_purifier_control.air_purifiers[
```

Comment:
Maybe we only need to pass the coordinator data to the lambda instead of the whole entity.